library/core/src/ptr/const_ptr.rs

```diff
@@ -2206,4 +2249,57 @@ mod verify {
         check_const_offset_from_tuple_4,
         check_const_offset_from_tuple_4_arr
     );
+
+    trait TestTrait {}
+
+    struct TestStruct {
+        value: i64,
+    }
+
+    impl TestTrait for TestStruct {}
+
+    macro_rules! gen_const_byte_arith_harness_for_dyn {
+        (byte_offset, $proof_name:ident) => {
+            // Workaround: Directly verifying the method `<*const dyn TestTrait>::byte_offset`
+            // causes a compilation error: "Failed to resolve checking function <*const dyn TestTrait>::byte_offset
+            // because Expected a type, but found trait object paths `dyn TestTrait`".
+            // As a result, the proof is annotated for the underlying struct type instead.
```

Comment:
Do you have a GitHub issue in Kani describing this failure? If so, could you add it here?

Reply:
Yes, the issue is [here](https://github.com/model-checking/kani/issues/3763), it has also been added to code comments.